src/elements/kw-social-comment/kw-social-comment.html

```diff
@@ -148,37 +173,77 @@ <h3 shown$="[[!comments.length]]">Be the first to comment!</h3>
-                </div>
-                <div class="content">
-                    <p><span class="bold">{{comment.author.username}}</span><span inner-h-t-m-l="[[_lb(comment.text)]]"></span></p>
-                    <date-util format="H:mm , dd Mon yyyy" date="[[_createDate(comment.date_created)]]"></date-util>
-                </div>
-            </div>
-        </template>
+        <iron-list items="[[comments]]" as="comment" hidden$="[[comment]]">
+                <template>
+                    <div class$="comment [[_computeOptimisticClass(comment.optimistic)]]">
+                        <div class="comment-avatar">
+                            <iron-image preload placeholder="/assets/icons/shares/commentAvatar.svg" fade sizing="contain" src="{{comment.author.avatar.urls.circle}}"></iron-image>
+                        </div>
+                        <div class="content">
+                            <p><span class="bold">{{comment.author.username}}</span><span inner-h-t-m-l="[[_lb(comment.text)]]"></span></p>
+                            <date-util format="H:mm , dd Mon yyyy" date="[[_createDate(comment.date_created)]]"></date-util>
+                        </div>
+                        <button id="retry" on-tap="_retryButtonTapped" hidden$="[[!comment.error]]" class="error">retry</button>
+                    </div>
+                </template>
+        </iron-list>
+        <button hidden$="[[!comments.length]]" class="loader" id="loader" on-tap="_loadMoreData">Load more...</button>
+
     </template>
     <script>
         Polymer({
             is: 'kw-social-comment',
+            behaviors: [Kano.APIClient],
             properties: {
                 comments: {
                     type: Array,
                     value: null
+                },
+                itemId: {
+                    type: String
+                },
+                pageNumber: {
+                    type: Number,
+                    value: 0,
+                    observer: "_onDataLoad"
+                }
+            },
+            ready () {
+            },
+            _onDataLoad () {
+                if (this.pageNumber === null) {
+                    this.$.loader.style.display = "none";
+                } else {
+                    this.$.loader.disabled = false;
+                    this.$.loader.style.background = "#ff842a";
                 }
             },
+            _loadMoreData (e) {
+                // When we hit the last page
+                if (this.pageNumber === null || !this.itemId) {
+                    return;
+                }
+                this.$.loader.disabled = true;
+                this.$.loader.style.background = "#999";
+                this.fire('load-comment', { id:this.itemId, page:this.pageNumber });
+            },
             _createDate (formatted) {
                 return new Date(formatted);
             },
+            _retryButtonTapped () {
+                this.$.retry.style.display = "none";
```

Comment:
I think all styles you are setting with JS could be refactored by using properties + css as mentioned above